### PR TITLE
feat(core): persistent login table records get removed after expiration

### PR DESCRIPTION
**Pull request by jeabakker**
The persistent login codes get removed from the database after the
corresponding cookie has expired.

EDIT:
also added cleanup of the remember_me table when a user gets removed

### CODE REVIEW

**Comment by hypeJunction** [comment]
The tests are probably failing because elgg() is getting called before plugins are bootstraped.

**Comment by jeabakker** [comment]
Failure probably has to do with https://travis-ci.org/Elgg/Elgg/jobs/393663732#L832, but i do't get (yet) how i changed something that would make that fail

**Comment by hypeJunction** [comment]
`elgg()` gets called too early, before plugins are booted, so that bootstraps `_elgg_services()->dic` without the plugin files, so subsequently in `system_log/start.php` it fatals, because service container was created/cached too early and there is no record of `system_log`

**Comment by jeabakker** [comment]
On my system `elgg()` doesn't get called before the plugins are being loaded

**Comment by jdalsem** [comment]
This fixes #11794

**Comment by jeabakker** [comment]
finally got everything working, ready for review ;)

File: engine/classes/Elgg/PersistentLoginService.php [hunk 1]
```diff
@@ -158,10 +160,15 @@ public function getUserFromHash($hash) {
 			return null;
 		}
 
-		$hash = $this->db->sanitizeString($hash);
-		$query = "SELECT guid FROM {$this->table} WHERE code = '$hash'";
+		
+		$query = "SELECT guid
```
Comment (hypeJunction) [inline engine/classes/Elgg/PersistentLoginService.php]: Why not use QueryBuilder?
```php
$qb = Select::fromTable($this->table);
$qb->select('guid')
       ->where($qb->compare('code', '=', $hash, 'string'));
```

And you can do same for `Update`, `Delete`, `Insert`. Ultimately, I think it's more readable and less prone to typos. Also, if you ever need to do a unit test, it's easier to mock queries.

**Review by hypeJunction — COMMENTED** [review]
This needs integration tests. I don't think we can rely on SQL query regex (they are good to have, but they don't test anything, if the query is flawed by design, it will pass the tests, but won't catch actual problems).

**Pull request merged**